bookstore-quarkus/src/main/kotlin/com/arconsis/data/books/BooksDataStore.kt

```diff
@@ -2,33 +2,49 @@ package com.arconsis.data.books
 
 import com.arconsis.http.books.Book
 import com.arconsis.http.books.CreateBook
+import io.quarkus.hibernate.reactive.panache.common.WithSession
+import io.quarkus.hibernate.reactive.panache.common.WithTransaction
 import io.smallrye.mutiny.Uni
 import io.smallrye.mutiny.coroutines.awaitSuspending
+import jakarta.enterprise.context.ApplicationScoped
 import java.util.*
-import javax.enterprise.context.ApplicationScoped
 
 @ApplicationScoped
 class BooksDataStore(private val booksRepository: BooksRepository) {
 
     suspend fun getBook(bookId: UUID): Book? {
-        val bookEntity = booksRepository.findById(bookId)?.awaitSuspending()
-        return bookEntity?.toBook()
+        return getBookEntity(bookId)
+            ?.awaitSuspending()
+            ?.toBook()
     }
 
+    @WithSession
+    protected fun getBookEntity(bookId: UUID): Uni<BookEntity>? {
```

Comment:
I assume this method needs to be `protected` because of the `@WithSession` annotation?

Reply:
yes, exactly. Annotations can only be on methods that can be overwritten.